src/KeyDB.php

```diff
@@ -73,37 +74,41 @@ public static function getSupportedKeyTypes(): array
     }
 
     /**
-     * Computes the hashed keys IDs for the given key metadata.
+     * Computes the key ID for the given key metadata.
+     *
+     * Per specification section 4.2, the KEYID is a hexdigest of the SHA-256
+     * hash of the canonical form of the key.
      *
      * @param \ArrayAccess $keyMeta
      *     An ArrayAccess object of key metadata. See self::addKey() and the TUF
      *     specification for the array structure.
      *
-     * @return string[]
-     *     An array of hashed key IDs for the key metadata. Each entry is a
-     *     string hash of the key signature and all its associated metadata.
-     *     There is one entry for each hashing algorithm specified in the
-     *     'keyid_hash_algorithms' child array.
+     * @return string
+     *     The key ID in hex format for the key metadata hashed using sha256.
      *
      * @see https://github.com/theupdateframework/specification/blob/v1.0.9/tuf-spec.md#4-document-formats
      *
      * @todo https://github.com/php-tuf/php-tuf/issues/56
      */
-    private static function computeKeyIds(\ArrayAccess $keyMeta): array
+    private static function computeKeyId(\ArrayAccess $keyMeta): string
     {
+        // @see https://github.com/secure-systems-lab/securesystemslib/blob/master/securesystemslib/keys.py
+        // The keyid_hash_algorithms array value is based on the TUF settings,
+        // it's not expected to be part of the key metadata. The fact that it is
+        // currently included is a quirk of the TUF python code that may be
+        // fixed in future versions. Calculate using the normal TUF settings
+        // since this is how it's calculated in the securesystemslib code and
+        // any value for keyid_hash_algorithms in the key data in root.json is
+        // ignored.
         $keyCanonicalStruct = [
             'keytype' => $keyMeta['keytype'],
             'scheme' => $keyMeta['scheme'],
-            'keyid_hash_algorithms' => $keyMeta['keyid_hash_algorithms'],
+            'keyid_hash_algorithms' => ['sha256', 'sha512'],
```

Comment:
does this come from `securesystemslib.settings.HASH_ALGORITHMS` setting?

If so should this be a setting on our side too? Could securesystemslib setup to use different algorithms or say just `sha256` or just `sha512`?

Reply:
It's really coming from the settings the tuf python client sends into the securesystemslib, I think.

If this value ever changed, all keyids would change, so I don't think they would ever change it unless in a future version that's incompatible.  Honestly, it's just stupid that it's even there.

Reply:
I guess it meant if someone want to set up the python server side. When they are setting it up to use can they change this setting before they are start using it? If so when they need to set up our client differently?

Reply:
I don't think they'd ever set it up differently - if they did it would be incompatible with any standard installs of the TUF client.  This quirk is just kind of stupid, so I think have a hard-coded values represents essentially the truth.

Reply:
@pwolanin okay that makes sense. If for some reason it was setup differently our client would fail because the key id computation would not work, correct? I don't think that is bad thing. I just want to make sure  we would fail if the server is not set up as we expect.

Reply:
Added a follow up #179 to ensure that we only accept exact expected value for `keyid_hash_algorithms`. This mean if for some reason it was changed on server we would fail.